charts/api/templates/deployment-queue.yaml

```diff
@@ -1,32 +1,34 @@
+{{- range .Values.queue.queueNames }}
+---
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: {{ include "api.fullname" . }}-queue
+  name: {{ include "api.fullname" $ }}-queue-{{ . }}
   labels:
-{{ include "api.labels" . | indent 4 }}
+{{ include "api.labels" $ | indent 4 }}
 spec:
-  replicas: {{ .Values.replicaCount.queue }}
+  replicas: {{ $.Values.replicaCount.queue }}
   selector:
     matchLabels:
-      app.kubernetes.io/name: {{ include "api.name" . }}
-      app.kubernetes.io/instance: {{ .Release.Name }}
+      app.kubernetes.io/name: {{ include "api.name" $ }}
+      app.kubernetes.io/instance: {{ $.Release.Name }}
       app.kubernetes.io/component: queue
   template:
     metadata:
       labels:
-        app.kubernetes.io/name: {{ include "api.name" . }}
-        app.kubernetes.io/instance: {{ .Release.Name }}
+        app.kubernetes.io/name: {{ include "api.name" $ }}
+        app.kubernetes.io/instance: {{ $.Release.Name }}
         app.kubernetes.io/component: queue
     spec:
-      serviceAccountName: {{ include "api.fullname" . }}-defaultrole
-    {{- with .Values.imagePullSecrets }}
+      serviceAccountName: {{ include "api.fullname" $ }}-defaultrole
+    {{- with $.Values.imagePullSecrets }}
       imagePullSecrets:
         {{- toYaml . | nindent 8 }}
```

Comment:
I suppose this `.` does the intended thing here because of the  `with` (= it should indeed be a `.` and not `$`)? Just asking in case it's an oversight and I understand it wrong

Reply:
Yeah `with` shifts the context yet again, so there's no need to change the lines that are within other loop constructs.